### PR TITLE
feat: canvas 支持挂载 s2 实例

### PR DESCRIPTION
**Pull request by wuhaiyang**
### 👀 PR includes

✨ Feature

- [x] New feature

🎨 Enhance

- [ ] Code style optimization
- [ ] Refactoring
- [ ] Change the UI
- [ ] Improve the performance
- [ ] Type optimization

🐛 Bugfix

- [ ] Solve the issue and close #0

🔧 Chore

- [ ] Test case
- [ ] Docs / demos update
- [ ] CI / workflow
- [ ] Release version
- [ ] Other ()

### 📝 Description

 背景：行为回放工具消费。监听s2#onScroll 事件，上报行为数据，回放时，希望通过canvas.$sheetInstance 实例去 scrollWithAnimation 回放

### 🖼️ Screenshot

| Before | After |
| ------ | ----- |
| ❌      | ✅     |

### 🔍 Self-Check before the merge

- [ ] Add or update relevant docs.
- [ ] Add or update relevant demos.
- [ ] Add or update test case.
- [ ] Add or update relevant TypeScript definitions.

### CODE REVIEW

File: packages/s2-core/__tests__/unit/sheet-type/table-sheet-spec.ts [hunk 1]
```diff
@@ -218,4 +218,11 @@ describe('TableSheet Tests', () => {
   test('should get content height', () => {
     expect(s2.getContentHeight()).toEqual(120);
   });
+
+  test('get sheetInstance from canvas', () => {
+    const canvas = s2.getCanvasElement();
```
Comment (lijinke666) [inline packages/s2-core/__tests__/unit/sheet-type/table-sheet-spec.ts:223]: getCanvasElement() 的类型直接改成 `HTMLCanvasElement & { $sheetInstance: SpreadSheet }` ? 不用每次 as

File: packages/s2-core/src/sheet-type/spread-sheet.ts [hunk 2]
```diff
@@ -159,6 +159,18 @@ export abstract class SpreadSheet extends EE {
...
     this.registerIcons();
     this.setOverscrollBehavior();
+    this.mountSheetInstance();
+  }
+
+  private mountSheetInstance(destroy?: boolean) {
+    const canvas = this.getCanvasElement();
+    if (canvas) {
+      if (!destroy) {
+        canvas.$sheetInstance = this;
```
Comment (lijinke666) [inline packages/s2-core/src/sheet-type/spread-sheet.ts:169]: 加上私有标识是不是要好点 `__sheetInstance__`

File: packages/s2-core/src/sheet-type/spread-sheet.ts [hunk 3]
```diff
@@ -440,8 +452,8 @@ export abstract class SpreadSheet extends EE {
     this.store?.clear();
     this.destroyTooltip();
     this.clearCanvasEvent();
+    this.mountSheetInstance(true);
```
Comment (lijinke666) [inline packages/s2-core/src/sheet-type/spread-sheet.ts:455]: `mountSheetInstance(true)` 我觉得在 `destroy` 上下文里面有点怪, 直接 `delete canvas.$sheetInstance` 更直观, mountSheetInstance 只保留纯粹的挂载逻辑

File: packages/s2-core/__tests__/unit/sheet-type/table-sheet-spec.ts (hunk 1 above)
Comment (wuhaiyang) [reply packages/s2-core/__tests__/unit/sheet-type/table-sheet-spec.ts:223]: done

File: packages/s2-core/src/sheet-type/spread-sheet.ts (hunk 2 above)
Comment (wuhaiyang) [reply packages/s2-core/src/sheet-type/spread-sheet.ts]: done

File: packages/s2-core/src/sheet-type/spread-sheet.ts (hunk 3 above)
Comment (wuhaiyang) [reply packages/s2-core/src/sheet-type/spread-sheet.ts]: done

**Review by lijinke666 — APPROVED** [review]

**Pull request merged**